docs/DeveloperGuide.adoc

```diff
@@ -235,6 +235,201 @@ image::CommitActivityDiagram.png[]
+* `MergeCommand#onlyOneMergeLeft()` -- Checks whether there is only one merge left.
+
+The implementation of `MergePolicyCommand` is similar.
+
+The child classes of `MergeConfirmedCommand` are `MergePersonConfirmedCommand` and `MergePolicyConfirmedCommand`, while the child classes of `MergeRejectedCommand` are `MergePersonRejectedCommand` and `MergePolicyRejectedCommand`.
+They all implement `#execute(Model)`. Additionally, these classes implement an `#isLastMerge()` command to indicate if this is the last possible merge for the entity being merged.
+
+AddressBookParser stores a boolean flag to indicate whether a merge is currently taking place. When it is set as true, all other commands will not be parsed and will be treated as invalid commands.
+The AddressBookParser object also stores the MergeCommand object during a merge process. This object is then used by `MergeConfirmedCommand` objects and `MergeRejectedCommand` objects in their execution.
+
+image::MergeClassDiagram.png[]
+
+Given below is an example usage scenario and how the merge mechanism behaves at each step.
+
+Step 1. The user adds a duplicate profile. The `AddCommand` will throw a `DuplicatePersonWithMergeException` during its execution.
+This exception is thrown if there is at least one different field between the input person and the original person stored in the model. Else, a `DuplicatePersonWithoutMergeException` will be thrown.
+The `DuplicatePersonWithMergeException` will finally be caught in the `CommandBox`. UI outputs the error message and a prompt to start a merge. `CommandBox` then constructs two command strings: one to proceed with the merge and one to reject the merge.
+This is done via `#standByForMerge(String, String)`. This string is then stored.
+
+Step 2. The user inputs yes or presses enter to proceed with the merge. `CommandBox` then calls `CommandExecutor#execute()` to execute the merge command it constructed previously.
+When the command is being parsed in the `AddressBookParser` object, a new `MergeCommand` object is created and stored. The `isMerging` flag is also set to true.
+The execution of this command then returns a CommandResult that prompts the next merge.
+
+Step 3. The user inputs yes or presses enter to update the field that was displayed in the prompt. The `AddressBookParser` parses the input and creates a new `MergePersonConfirmedCommand` object.
+The `MergePersonConfirmedCommand` object obtains information for the merge from the `MergeCommand` object that was passed in as a parameter in the constructor.
+In the execution, a new `EditCommand` is created and `EditCommand#executeForMerge()` is used to update the person in the model. If the
+`MergePersonConfirmedCommand#isLastMerge` returns false, `MergeCommand#removeFirstDifferentField` is called and the command result then shows a success message and the next prompt.
+
+This process is shown in the sequence diagram below.
+
+image::MergeSequenceDiagram.png[]
```

Comment:
You may want to consider using a better color scheme and larger font such that it is easier to read and follow.